software-mansion/react-native-reanimated
# [LA Web] Fix animations in production bundle

## Summary

As pointed in #5561, layout animations don't work with production bundle. It is caused by fragile method of obtaining animation name, i.e. using `config.name` if config is function. During bundling, name of this function may change (in linked issue to `c`, in my case to `r`), so instead of animation user will see warning that reanimated couldn't load animation. 

This PR adds static field `className` to animations builders in order to provide access to animation name, even if config name gets changed during bundling.  

## Test plan

1.  Create new app (`npx create-expo-app`)
2. Install Reanimated from package (`npm pack` -> `yarn add <path_to_package>`) 
3. `npx expo export -p web`
4. `npx serve dist --single`

I used reanimated from local package to make sure that `expo export` works correctly.

<details>
<summary> Tested on the following code </summary>

```jsx
import Animated, { FadeOut, FadeOutRight } from "react-native-reanimated";
import { View, Button } from "react-native";
import { useState } from "react";

export default function AnimatedStyleUpdateExample(props) {
  const [show, setShow] = useState(true);

  return (
    <View
      style={{
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      {show && (
        <Animated.View
          style={[{ width: 100, height: 80, backgroundColor: "black" }]}
          exiting={FadeOut}
        />
      )}

      <Button
        title="toggle"
        onPress={() => {
          setShow((prev) => !prev);
        }}
      />
    </View>
  );
}

```

</details>
